sid/script.js

```diff
@@ -0,0 +1,633 @@
+}
+
+function checkFullSquares() {
+    for (const block of allBlocks) {
+        if (block.textContent !== '') {
+            block.style.border = '#fcfcfc solid 4px';
+        }
+        else {
+            block.style.border = 'grey solid 1px';
+        }
+    }
+}
+
+function wordInDict(word) {
+    if (dictionary.includes(word)) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+function wordFound(event) {
+    if (gameComplete === false) {
+        if (event.type === 'click') {
+            wordTypeOut(event.target.textContent);
+        }
+        if (event.type === 'keyup' && event.key.length === 1 && event.key.match(/[a-z]/i) || event.key === 'Enter' || event.key === 'Backspace') {
+            buttonPress(event.key);
+            wordTypeOut(event.key);
+        }
+    }
+}
+
+
+//inspiration from An Object Is A's Medium Article on making images pop in and pop out
+function imageFadeIn(image) {
+    image.style.transitionDuration = '0.2s';
+    image.style.transform = 'scale(1.5)';
+
+    setTimeout(function () {
+        image.style.transitionDuration = '0.07s';
+        image.style.transform = 'scale(1)';
+    }, 200);
+}
+
+function imageFadeOut(image) {
+    image.style.transitionDuration = '0.2s';
+    image.style.transform = 'scale(1.5)';
+
+    setTimeout(function () {
+        image.style.transitionDuration = '0.07s';
+        image.style.transform = 'scale(0)';
+        image.src = '';
+    }, 200);
+
+}
+
+function changeButtonColor(button, color) {
+    if (color === 'green' || color === 'grey') {
+        button.style.color = 'white';
+    }
+    else {
+        button.style.color = 'black';
+    }
+    if (button.style.backgroundColor !== 'green') {
+        if (color === 'grey') {
+            if (button.style.backgroundColor !== 'yellow') {
+                button.style.backgroundColor = 'grey';
+            }
+        }
+        else {
+            button.style.backgroundColor = color;
+        }
+    }
+}
+
+function addTwoRows() {
+    rowsAdded = true;
+    if (gameComplete === false) {
+        if (rowNumber <= 6) {
+            let number = 2;
+            let square = 5;
+            while (number !== 0) {
+                square = 5;
+                const newRow = document.createElement('div');
+                newRow.classList.add('row');
+                newRow.classList.add('newRow');
+                while (square > 0) {
+                    const newSquare = document.createElement('div');
+                    newSquare.classList.add('squares');
+                    newRow.appendChild(newSquare);
+                    square -= 1;
+                }
+                grid.appendChild(newRow);
+                number -= 1;
+            }
+            moreChances.style.display = 'none';
+            score -= 10;
+            allBlocks = document.querySelectorAll('.squares');
+            rows = document.querySelectorAll('.row');
+        }
+    }
+}
```

Comment:
Maybe, this would make it easier to read?

```suggestion
function addTwoRows() {
    rowsAdded = true;
    
    if (gameComplete || rowNumber > 6) {
        return;
    }
    
    let number = 2;
    let square = 5;
    
    while (number !== 0) {
        square = 5;
        const newRow = document.createElement('div');
        newRow.classList.add('row');
        newRow.classList.add('newRow');
        while (square > 0) {
            const newSquare = document.createElement('div');
            newSquare.classList.add('squares');
            newRow.appendChild(newSquare);
            square -= 1;
        }
        grid.appendChild(newRow);
        number -= 1;
    }
    moreChances.style.display = 'none';
    score -= 10;
    allBlocks = document.querySelectorAll('.squares');
    rows = document.querySelectorAll('.row');
}
```